### PR TITLE
Scenario workflow for a production recommended deployment

### PR DESCRIPTION
**Pull request by rohit2b**
Confluent recommends these security mechanisms for a production deployment:
- Enable Kafka client authentication. Choose one of:
  - SASL/Plain or mTLS
  - For SASL/Plain, the identity can come from LDAP server
- Enable Confluent Role Based Access Control for authorization, with user/group identity coming from LDAP server
- Enable TLS for network encryption - both internal (between CP components) and external (Clients to CP components)

In this deployment scenario, we will set this up, choosing SASL/Plain for authentication, using user provided custom certificates.

### CODE REVIEW

File: production-secure-deploy/README.rst [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+* RBAC principal credentials
+  
+You can either provide your own certificates, or generate test certificates. Follow instructions
+in the below "Appendix: Create your own certificates" section to see how to generate certificates
+and set the appropriate SANs. 
+
+Provide component TLS certificates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+   ::
```
Comment (arodoni) [inline production-secure-deploy/README.rst:114]:
```suggestion
::
```

File: production-secure-deploy/README.rst [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+   
+     kubectl create secret generic tls-group1 \
+      --from-file=fullchain.pem=$TUTORIAL_HOME/../assets/certs/generated/server.pem \
```
Comment (arodoni) [inline production-secure-deploy/README.rst:117]:
```suggestion
   --from-file=fullchain.pem=$TUTORIAL_HOME/../assets/certs/generated/server.pem \
```

File: production-secure-deploy/README.rst [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+      --from-file=cacerts.pem=$TUTORIAL_HOME/../assets/certs/generated/ca.pem \
```
Comment (arodoni) [inline production-secure-deploy/README.rst:118]:
```suggestion
   --from-file=cacerts.pem=$TUTORIAL_HOME/../assets/certs/generated/ca.pem \
```

File: production-secure-deploy/README.rst [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+      --from-file=privkey.pem=$TUTORIAL_HOME/../assets/certs/generated/server-key.pem
```
Comment (arodoni) [inline production-secure-deploy/README.rst:119]:
```suggestion
     kubectl create secret generic tls-group1 \
      --from-file=fullchain.pem=$TUTORIAL_HOME/../assets/certs/generated/server.pem \
      --from-file=cacerts.pem=$TUTORIAL_HOME/../assets/certs/generated/ca.pem \
      --from-file=privkey.pem=$TUTORIAL_HOME/../assets/certs/generated/server-key.pem
```
```suggestion
     kubectl create secret generic tls-group1 \
      --from-file=fullchain.pem=$TUTORIAL_HOME/../assets/certs/generated/server.pem \
      --from-file=cacerts.pem=$TUTORIAL_HOME/../assets/certs/generated/ca.pem \
      --from-file=privkey.pem=$TUTORIAL_HOME/../assets/certs/generated/server-key.pem
```

File: production-secure-deploy/README.rst [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Create Kubernetes secret objects for MDS:
+
+   ::
+   
+     kubectl create secret generic mds-token \
+       --from-file=mdsPublicKey.pem=$TUTORIAL_HOME/../assets/certs/mds-publickey.txt \
+       --from-file=mdsTokenKeyPair.pem=$TUTORIAL_HOME/../assets/certs/mds-tokenkeypair.txt
+
```
Comment (arodoni) [inline production-secure-deploy/README.rst:155]:
```suggestion
   
   #. Create a Kubernetes secret object for Control Center:
```

File: production-secure-deploy/README.rst [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+      --from-file=ldap.txt=$TUTORIAL_HOME/ldap.txt
+
+   In this tutorial, we use one credential for authenticating all client and
+   server communication to Kafka brokers. In production scenarios, you'll want
+   to specify different credentials for each of them.
+
+Provide RBAC principal credentials
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Create Kubernetes secret objects for MDS:
```
Comment (arodoni) [inline production-secure-deploy/README.rst:148]:
```suggestion
#. Create a Kubernetes secret object for MDS:
```

**Review by arodoni — APPROVED** [review]
With a few formatting fix suggestions.

File: production-secure-deploy/README.rst (hunk 5 above)
Comment (rohit2b) [reply production-secure-deploy/README.rst:155]: Added clarifications here

**Comment by rohit2b** [comment]
Thanks! Incorporated formatting suggestions.

**Pull request merged**